utilities/src/main/java/io/syndesis/qe/templates/MongoDb36Template.java

```diff
@@ -0,0 +1,132 @@
+        accountParameters.put("host", APP_NAME);
+        accountParameters.put("user", MONGODB_USER);
+        accountParameters.put("password", MONGODB_PASSWORD);
+        accountParameters.put("database", MONGODB_DATABASE);
+        // this does not work for now
+        //        accountParameters.put("admindb", MONGODB_DATABASE);
+        accountParameters.put("url", MONGDB_URL);
+        mongodbAccount.setProperties(accountParameters);
+        AccountsDirectory.getInstance().addAccount(Account.Name.MONGODB36.getId(), mongodbAccount);
+
+        if (TestUtils.isDcDeployed(APP_NAME)) {
+            return;
+        }
+
+        List<ContainerPort> ports = new LinkedList<>();
+        ports.add(new ContainerPortBuilder()
+            .withName("mongodb")
+            .withContainerPort(MONGODB_PORT)
+            .withProtocol("TCP").build());
+
+        List<EnvVar> templateParams = new ArrayList<>();
+        templateParams.add(new EnvVar("MONGODB_ADMIN_PASSWORD", ADMIN_PASSWORD, null));
+        templateParams.add(new EnvVar("MONGODB_USER", MONGODB_USER, null));
+        templateParams.add(new EnvVar("MONGODB_PASSWORD", MONGODB_PASSWORD, null));
+        templateParams.add(new EnvVar("MONGODB_DATABASE", MONGODB_DATABASE, null));
+
+        OpenShiftUtils.getInstance().deploymentConfigs().createOrReplaceWithNew()
+            .editOrNewMetadata()
+            .withName(APP_NAME)
+            .addToLabels(LABEL_NAME, APP_NAME)
+            .endMetadata()
+
+            .editOrNewSpec()
+            .addToSelector(LABEL_NAME, APP_NAME)
+            .withReplicas(1)
+            .editOrNewTemplate()
+            .editOrNewMetadata()
+            .addToLabels(LABEL_NAME, APP_NAME)
+            .endMetadata()
+            .editOrNewSpec()
+            .addNewContainer().withName(APP_NAME).withImage(MONGODB_IMAGE)
+            .addAllToPorts(ports)
+            .addAllToEnv(templateParams)
+
+            .endContainer()
+            .endSpec()
+
+            .endTemplate()
+            .addNewTrigger()
+            .withType("ConfigChange")
+            .endTrigger()
+            .endSpec()
+            .done();
+
+        ServiceSpecBuilder serviceSpecBuilder = new ServiceSpecBuilder().addToSelector(LABEL_NAME, APP_NAME);
+
+        serviceSpecBuilder.addToPorts(new ServicePortBuilder()
+            .withName("mongodb")
+            .withPort(MONGODB_PORT)
+            .withTargetPort(new IntOrString(MONGODB_PORT))
+            .build());
+
+        OpenShiftUtils.getInstance().services().createOrReplaceWithNew()
+            .editOrNewMetadata()
+            .withName(APP_NAME)
+            .addToLabels(LABEL_NAME, APP_NAME)
+            .endMetadata()
+            .editOrNewSpecLike(serviceSpecBuilder.build())
+            .endSpec()
+            .done();
+
+        try {
+            OpenShiftWaitUtils.waitFor(OpenShiftWaitUtils.areExactlyNPodsReady(LABEL_NAME, APP_NAME, 1));
+            Thread.sleep(20 * 1000);
```

Comment:
why this sleep?

Reply:
honestly don't know, most of this is copy-pasta from other templates, will remove